### PR TITLE
Update fedora distro info

### PR DESCRIPTION
**Pull request by enriqueartal**
This PR aims to be a continuation of #37833 in order to add more possible system packages and correct some other ones in `build/pkgs/*/distros/fedora.txt`. After a fresh installation I got some errors:

- `src/sage/rings/lazy_series_ring.py`. The test passess without the PR. I am not sure which package is the cause.
```
File "src/sage/rings/lazy_series_ring.py", line 2008, in sage.rings.lazy_series_ring.LazyPowerSeriesRing.__init__
Failed example:
    TestSuite(L).run(skip=['_test_revert'])
Expected nothing
Got:
    Failure in _test_associativity:
    Traceback (most recent call last):
      File "/home/artal/sage/src/sage/misc/sage_unittest.py", line 298, in run
        test_method(tester=tester)
      File "/home/artal/sage/src/sage/categories/semigroups.py", line 123, in _test_associativity
        tester.assertEqual((x * y) * z, x * (y * z))
      File "/usr/lib64/python3.12/unittest/case.py", line 885, in assertEqual
        assertion_func(first, second, msg=msg)
      File "/usr/lib64/python3.12/unittest/case.py", line 878, in _baseAssertEqual
        raise self.failureException(msg)
    AssertionError: s + ([61 chars]^4*t+2*s^2*t^3+5*s*t^4) + (s^7+3*s^5*t^2+2*s^4[27 chars],t)^8 != s + ([61 chars]^4*t+4*s^3*t^2+2*s^2*t^3+5*s*t^4) + (s^7+3*s^5[37 chars],t)^8
    ------------------------------------------------------------
    Failure in _test_distributivity:
    Traceback (most recent call last):
      File "/home/artal/sage/src/sage/misc/sage_unittest.py", line 298, in run
        test_method(tester=tester)
      File "/home/artal/sage/src/sage/categories/distributive_magmas_and_additive_magmas.py", line 83, in _test_distributivity
        tester.assertEqual((x + y) * z, (x * z) + (y * z))
      File "/usr/lib64/python3.12/unittest/case.py", line 885, in assertEqual
        assertion_func(first, second, msg=msg)
      File "/usr/lib64/python3.12/unittest/case.py", line 878, in _baseAssertEqual
        raise self.failureException(msg)
    AssertionError: 2 + ([60 chars]3) + (s^5+5*s^4*t+4*s^3*t^2+4*s^2*t^3+5*s*t^4+[36 chars],t)^7 != 2 + ([60 chars]3) + 4*s^2*t^2 + (s^5+5*s^4*t+4*s^3*t^2+4*s^2*[48 chars],t)^7
    ------------------------------------------------------------
    The following tests failed: _test_associativity, _test_distributivity
**********************************************************************
1 item had failures:
   1 of  33 in sage.rings.lazy_series_ring.LazyPowerSeriesRing.__init__
    [685 tests, 1 failure, 15.80 s]
----------------------------------------------------------------------
sage -t --warn-long 22.2 --random-seed=128527404001182411276111307855871193848 src/sage/rings/lazy_series_ring.py  # 1 doctest failed
----------------------------------------------------------------------
Total time for all tests: 16.6 seconds
    cpu time: 15.8 seconds
    cumulative wall time: 15.8 seconds
Features detected for doctesting: lrcalc_python,sage.libs.flint,sage.libs.pari,sage.libs.singular,sage.modules,sage.rings.finite_rings,sage.rings.number_field,sage.symbolic
```
- `src/sage/rings/polynomial/multi_polynomial_ideal.py`. Apparently `cddlib` is causing it.
```
File "src/sage/rings/polynomial/multi_polynomial_ideal.py", line 4192, in sage.rings.polynomial.multi_polynomial_ideal.NCPolynomialIdeal.groebner_fan
Failed example:
    g.reduced_groebner_bases()
Exception raised:
    Traceback (most recent call last):
      File "/home/artal/sage/src/sage/rings/polynomial/groebner_fan.py", line 1064, in reduced_groebner_bases
        return self.__reduced_groebner_bases
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    AttributeError: 'GroebnerFan' object has no attribute '_GroebnerFan__reduced_groebner_bases'. Did you mean: '_GroebnerFan__is_groebner_basis'?

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/home/artal/sage/src/sage/rings/polynomial/groebner_fan.py", line 1015, in _gfan_reduced_groebner_bases
        return self.__gfan_reduced_groebner_bases
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    AttributeError: 'GroebnerFan' object has no attribute '_GroebnerFan__gfan_reduced_groebner_bases'

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/home/artal/sage/src/sage/doctest/forker.py", line 715, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/home/artal/sage/src/sage/doctest/forker.py", line 1147, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.rings.polynomial.multi_polynomial_ideal.NCPolynomialIdeal.groebner_fan[3]>", line 1, in <module>
        g.reduced_groebner_bases()
      File "/home/artal/sage/src/sage/rings/polynomial/groebner_fan.py", line 1066, in reduced_groebner_bases
        G = self._gfan_reduced_groebner_bases()
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/artal/sage/src/sage/rings/polynomial/groebner_fan.py", line 1017, in _gfan_reduced_groebner_bases
        B = self.gfan(cmd='bases')
            ^^^^^^^^^^^^^^^^^^^^^^
      File "/home/artal/sage/src/sage/rings/polynomial/groebner_fan.py", line 1139, in gfan
        s = gfan(I, cmd, verbose=self.__verbose)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/artal/sage/src/sage/misc/decorators.py", line 658, in wrapper
        return func(*args, **kwds)
               ^^^^^^^^^^^^^^^^^^^
      File "/home/artal/sage/src/sage/interfaces/gfan.py", line 119, in __call__
        raise RuntimeError(err)
    RuntimeError: Falling back on CddLib
    Falling back on CddLib

**********************************************************************
1 item had failures:
   1 of   5 in sage.rings.polynomial.multi_polynomial_ideal.NCPolynomialIdeal.groebner_fan
    [984 tests, 1 failure, 5.50 s]
----------------------------------------------------------------------
sage -t --warn-long 22.2 --random-seed=196383766288019021986381945645510832135 src/sage/rings/polynomial/multi_polynomial_ideal.py  # 1 doctest failed
----------------------------------------------------------------------
Total time for all tests: 11.3 seconds
    cpu time: 5.3 seconds
    cumulative wall time: 5.5 seconds
Features detected for doctesting: sage.combinat,sage.libs.flint,sage.libs.singular,sage.modules,sage.plot,sage.rings.finite_rings,sage.rings.number_field,sage.symbolic
```
- `src/sage/interfaces/maxima_abstract.py`. The problem seems to be `maxima`. 
```
File "src/sage/interfaces/maxima_abstract.py", line 161, in sage.interfaces.maxima_abstract.MaximaAbstract._command_runner
Failed example:
    maxima._command_runner('describe', 'gcd')
Expected:
    -- Function: gcd (<p_1>, <p_2>, <x_1>, ...)
    ...
Got:
    <BLANKLINE>
    <BLANKLINE>
     -- Función: gcd (<p_1>, <p_2>, <x_1>, ...)
         Devuelve el máximo común divisor de <p_1> y <p_2>.  La variable
         ‘gcd’ determiona qué algoritmo se va a utilizar.  Asignándole a
         ‘gcd’ los valores ‘ez’, ‘subres’, ‘red’ o ‘spmod’, se seleccionan
         los algoritmos ‘ezgcd’, subresultante ‘prs’, reducido o modular,
         respectivamente.  Si ‘gcd’ vale ‘false’ entonces ‘gcd(<p_1>, <p_2>,
         <x>)’ devolverá siempre 1 para cualquier <x>.  Muchas funciones
         (por ejemplo, ‘ratsimp’, ‘factor’, etc.)  hacen uso de ‘gcd’
         implícitamente.  En caso de polinomios homogéneos se recomienda
         darle a ‘gcd’ el valor ‘subres’.  Para calcular un máximo común
         divisor en presencia de raíces, como en ‘gcd (<x>^2 - 2*sqrt(2)*<x>
         + 2, <x> - sqrt(2))’, la variable ‘algebraic’ debe igualarse a
         ‘true’ y ‘gcd’ no puede ser ‘ez’.
    <BLANKLINE>
         Se recomienda utilizar el algoritmo ‘subres’ en lugar de ‘red’, por
         ser aquél más moderno.
    <BLANKLINE>
         Si la variable ‘gcd’, cuyo valor por defecto es ‘spmod’, vale
         ‘false’, no se calculará el máximo común divisor cuando las
         expresiones se conviertan a su forma canónica (CRE), lo que
         redundará en ocasiones en mayor rapidez de cálculo.
    <BLANKLINE>
      There are also some inexact matches for `gcd'.
      Try `?? gcd' to see them.
    <BLANKLINE>
                                         true
    <BLANKLINE>
**********************************************************************
File "src/sage/interfaces/maxima_abstract.py", line 198, in sage.interfaces.maxima_abstract.MaximaAbstract.help
Failed example:
    maxima.help('gcd')
Expected:
    -- Function: gcd (<p_1>, <p_2>, <x_1>, ...)
    ...
Got:
    <BLANKLINE>
    <BLANKLINE>
     -- Función: gcd (<p_1>, <p_2>, <x_1>, ...)
         Devuelve el máximo común divisor de <p_1> y <p_2>.  La variable
         ‘gcd’ determiona qué algoritmo se va a utilizar.  Asignándole a
         ‘gcd’ los valores ‘ez’, ‘subres’, ‘red’ o ‘spmod’, se seleccionan
         los algoritmos ‘ezgcd’, subresultante ‘prs’, reducido o modular,
         respectivamente.  Si ‘gcd’ vale ‘false’ entonces ‘gcd(<p_1>, <p_2>,
         <x>)’ devolverá siempre 1 para cualquier <x>.  Muchas funciones
         (por ejemplo, ‘ratsimp’, ‘factor’, etc.)  hacen uso de ‘gcd’
         implícitamente.  En caso de polinomios homogéneos se recomienda
         darle a ‘gcd’ el valor ‘subres’.  Para calcular un máximo común
         divisor en presencia de raíces, como en ‘gcd (<x>^2 - 2*sqrt(2)*<x>
         + 2, <x> - sqrt(2))’, la variable ‘algebraic’ debe igualarse a
         ‘true’ y ‘gcd’ no puede ser ‘ez’.
    <BLANKLINE>
         Se recomienda utilizar el algoritmo ‘subres’ en lugar de ‘red’, por
         ser aquél más moderno.
    <BLANKLINE>
         Si la variable ‘gcd’, cuyo valor por defecto es ‘spmod’, vale
         ‘false’, no se calculará el máximo común divisor cuando las
         expresiones se conviertan a su forma canónica (CRE), lo que
         redundará en ocasiones en mayor rapidez de cálculo.
    <BLANKLINE>
      There are also some inexact matches for `gcd'.
      Try `?? gcd' to see them.
    <BLANKLINE>
                                         true
    <BLANKLINE>
**********************************************************************
2 items had failures:
   1 of   2 in sage.interfaces.maxima_abstract.MaximaAbstract._command_runner
   1 of   2 in sage.interfaces.maxima_abstract.MaximaAbstract.help
    [237 tests, 2 failures, 2.61 s]
----------------------------------------------------------------------
sage -t --random-seed=222682497766095343035925295278558065112 src/sage/interfaces/maxima_abstract.py  # 2 doctests failed
----------------------------------------------------------------------
Total time for all tests: 7.7 seconds
    cpu time: 1.8 seconds
    cumulative wall time: 2.6 seconds
Features detected for doctesting: sage.libs.pari
```
- `src/sage/interfaces/singular.py`. The problem is probably caused by `singular`:
```
File "src/sage/interfaces/singular.py", line 2326, in sage.interfaces.singular.SingularFunctionElement._instancedoc_
Failed example:
    'matrix_expression' in A.nrows.__doc__
Expected:
    True
Got:
    False
**********************************************************************
File "src/sage/interfaces/singular.py", line 2425, in sage.interfaces.singular.get_docstring
Failed example:
    'groebner' in get_docstring('groebner')
Expected:
    True
Got:
    False
**********************************************************************
File "src/sage/interfaces/singular.py", line 2427, in sage.interfaces.singular.get_docstring
Failed example:
    'standard.lib' in get_docstring('groebner')
Expected:
    True
Got:
    False
**********************************************************************
2 items had failures:
   1 of   4 in sage.interfaces.singular.SingularFunctionElement._instancedoc_
   2 of   4 in sage.interfaces.singular.get_docstring
    [412 tests, 3 failures, 1.78 s]
----------------------------------------------------------------------
sage -t --warn-long 22.2 --random-seed=305672045870174292260177177950580332722 src/sage/interfaces/singular.py  # 3 doctests failed
----------------------------------------------------------------------
Total time for all tests: 12.0 seconds
    cpu time: 0.6 seconds
    cumulative wall time: 1.8 seconds
Features detected for doctesting: 
```
- `src/sage/interfaces/giac.py`. The problem is caused by `giac`.
```
ile "src/sage/interfaces/giac.py", line 295, in sage.interfaces.giac.Giac
Failed example:
    I1=(1/(cos(2*y)+cos(y))).integral(y,0,pi/4).simplify()
Expected nothing
Got:
    Giac crashed -- automatically restarting.
**********************************************************************
File "src/sage/interfaces/giac.py", line 296, in sage.interfaces.giac.Giac
Failed example:
    (I1-((-2*ln((sqrt(3)-3*tan(1/8*pi))/(sqrt(3)+3*tan(1/8*pi)))*sqrt(3)-3*tan(1/8*pi))/9)).normal()
Expected:
    0
Got:
    Giac crashed -- automatically restarting.
    sage96
**********************************************************************
File "src/sage/interfaces/giac.py", line 298, in sage.interfaces.giac.Giac
Failed example:
    ((y+z*sqrt(5))*(y-sqrt(5)*z)).normal()
Exception raised:
    Traceback (most recent call last):
      File "/home/artal/sage/src/sage/doctest/forker.py", line 715, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/home/artal/sage/src/sage/doctest/forker.py", line 1147, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.interfaces.giac.Giac[8]>", line 1, in <module>
        ((y+z*sqrt(Integer(5)))*(y-sqrt(Integer(5))*z)).normal()
            ~^~~~~~~~~~~~~~~~~
      File "sage/structure/element.pyx", line 1512, in sage.structure.element.Element.__mul__
        return coercion_model.bin_op(left, right, mul)
      File "sage/structure/coerce.pyx", line 1236, in sage.structure.coerce.CoercionModel.bin_op
        return PyObject_CallObject(op, xy)
      File "sage/structure/element.pyx", line 1510, in sage.structure.element.Element.__mul__
        return (<Element>left)._mul_(right)
      File "sage/structure/element.pyx", line 1556, in sage.structure.element.Element._mul_
        return python_op(other)
      File "/home/artal/sage/src/sage/interfaces/interface.py", line 1600, in _mul_
        return self._operation('*', right)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/artal/sage/src/sage/interfaces/interface.py", line 1471, in _operation
        P = self._check_valid()
            ^^^^^^^^^^^^^^^^^^^
      File "/home/artal/sage/src/sage/interfaces/expect.py", line 1550, in _check_valid
        raise ValueError("The %s session in which this object was defined is no longer running." % P.name())
    ValueError: The giac session in which this object was defined is no longer running.
**********************************************************************
1 item had failures:
   3 of  15 in sage.interfaces.giac.Giac
    [182 tests, 3 failures, 2.10 s]
----------------------------------------------------------------------
sage -t --warn-long 22.2 --random-seed=159177512075622979747943355576924465951 src/sage/interfaces/giac.py  # 3 doctests failed
----------------------------------------------------------------------
Total time for all tests: 7.1 seconds
    cpu time: 0.5 seconds
    cumulative wall time: 2.1 seconds
Features detected for doctesting: 
```
- `src/sage/calculus/tests.py`.
```
File "src/sage/calculus/tests.py", line 126, in sage.calculus.tests
Failed example:
    integrate(x^x,x)
Expected:
    integrate(x^x, x)
Got:
    /usr/include/c++/14/bits/stl_vector.h:1130: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = long long int; _Alloc = std::allocator<long long int>; reference = long long int&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.
    integrate(x^x, x)
**********************************************************************
1 item had failures:
   1 of  79 in sage.calculus.tests
    [78 tests, 1 failure, 1.44 s]
----------------------------------------------------------------------
sage -t --warn-long 41.6 --random-seed=253346199774111986807301801043024986436 src/sage/calculus/tests.py  # 1 doctest failed
----------------------------------------------------------------------
Total time for all tests: 1.5 seconds
    cpu time: 2.0 seconds
    cumulative wall time: 1.4 seconds
Features detected for doctesting: 
```
- `src/sage/rings/polynomial/groebner_fan.py`. There are 47 failed doctests, caused by `gfan`.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Comment by enriqueartal** [comment]
After `.configure --without-system-cddlib --without-system-maxima --without-system-giac --without-system-gfan`, some errors are gone (`singular` spkg is also installed) but other appeared and I should recompile again.

**Comment by enriqueartal** [comment]
With these options, the errors are gone.

File: build/pkgs/cryptominisat/distros/fedora.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+cryptominisat
```
Comment (mkoeppe) [inline build/pkgs/cryptominisat/distros/fedora.txt]: This SPKG has actually been removed in the meantime.

**Comment by enriqueartal** [comment]
I guess there will be errors, I needed some `--without-system-___`. I had also some issues with `-enable-system-site-packages`; in particular that it took a differente `snappy`.

**Comment by mkoeppe** [comment]
Sure, but we see these failures (from newer giac etc.) also on other platforms

> I had also some issues with `-enable-system-site-packages`; in particular that it took a differente `snappy`.

Could you elaborate on that?

**Comment by mkoeppe** [comment]
Some of the newly added packages are not available on older `fedora` versions. See https://github.com/mkoeppe/sage/actions/runs/9970528956/job/27549645916#step:11:95
```
Error: Unable to find a match: gap-pkg-anupq gap-pkg-fplsa gap-pkg-permut gap-pkg-standardff
```

We'll have to adjust `tox.ini` for this

**Comment by mkoeppe** [comment]
New tests at https://github.com/mkoeppe/sage/actions/runs/9972513511

**Comment by mkoeppe** [comment]
Oops, new tests at https://github.com/mkoeppe/sage/actions/runs/9972676999

**Comment by mkoeppe** [comment]
Looking OK now for the `-standard` configurations. 
The experimental `-standard-sitepackages` configuration is only tested for `fedora-38` and `-39` and fails in building `fpylll` and `cypari2`. https://github.com/mkoeppe/sage/actions/runs/9972676999/job/27574846587#step:11:5050

**Comment by mkoeppe** [comment]
To fix this, we may need to tighten the version requirements for `cython` and/or `cysignals` (these are set in `src/pyproject.toml` since recently).

**Comment by mkoeppe** [comment]
> The experimental `-standard-sitepackages` configuration is only tested for `fedora-38` and `-39` and fails in building `fpylll` and `cypari2`

I switched it to testing `fedora-40`, and it fails in the same way: https://github.com/mkoeppe/sage/actions/runs/9981541731/job/27597777434#step:11:4753

**Comment by enriqueartal** [comment]
> Sure, but we see these failures (from newer giac etc.) also on other platforms
> 
> > I had also some issues with `-enable-system-site-packages`; in particular that it took a differente `snappy`.
> 
> Could you elaborate on that?

For more elaborated info I will test again to check what happened. Concerning `snappy` there is a packages `python3-snappy` in fedora but it is a compressor, and with `./configure --enable-system-site-packages`, it says it will use it, but it is not the very useful knot theory package.
The changes you are doing are above my expertise, I trust you.

**Comment by dimpase** [comment]
> > Sure, but we see these failures (from newer giac etc.) also on other platforms
> > 
> > > I had also some issues with `-enable-system-site-packages`; in particular that it took a differente `snappy`.
> > 
> > Could you elaborate on that?
> 
> For more elaborated info I will test again to check what happened. Concerning `snappy` there is a packages `python3-snappy` in fedora but it is a compressor, and with `./configure --enable-system-site-packages`, it says it will use it, but it is not the very useful knot theory package.

the name of the Fedora package does not matter.
What matters is the name of the Python module it installs.
If it happens to have the same module name as 
the knot theory snappy, one needs to do an extra test to tell them apart.
Not hard to fix.


> The changes you are doing are above my expertise, I trust you.

**Comment by dimpase** [comment]
It's not really Fedora-specific. The problem is that https://pypi.org/project/python-snappy/ installs Python module called `snappy`, and Fedora includes a version of the latter PyPI package as a package.

**Comment by mkoeppe** [comment]
Let's merge the present version and iterate based on it.

**Pull request merged**